tensorflow/python/ops/image_ops_impl.py

```diff
@@ -211,6 +211,45 @@ def _EnsureTensorIs4D(image):
 
   return (image, is_batch)
 
+def _flip_image(image, axis, random=False, seed=None):
+  """
+  Flips image(s) around a given axis.
+
+  Args:
+    image: 4-D Tensor of shape `[batch, height, width, channels]` or
+           3-D Tensor of shape `[height, width, channels]`.
+
+    axis:  A Tensor. Must be one of the following types: int32, int64. 1-D.
```

Comment:
Why 1D? This should be a scalar? 

I understand you can pass 1D to reverse, but unless you use that, I would keep it a scalar here, and pass [axis] to reverse. 

If you want to keep it 1D, you should change the docstring to explain that each components needs to be in the range given, and possibly they have to be ordered? Not sure. Definitely adds complexity you don't really need.

Reply:
I agree that it should be a scalar.

One strange thing about this refactor is that it forces the caller to provide an axis relative to a 4-D Tensor, even if they've passed in a 3-D image.

For example `random_flip_up_down()` needs to provide an axis, but doesn't know whether or not the image it's working with is 3-D or a 4-D batch of images. Currently I just pass in `1` with the knowledge that a single image will be extended to 4 dimensions.